lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -2100,10 +2116,11 @@ void ModuleEmitter::collectNamesEmitDecls(Block &block) {
     assert(op.getNumResults() == 1 && "firrtl only has single-op results");
     Value result = op.getResult(0);
 
-    // If this is an expression emitted inline or unused, it doesn't need a
-    // name.
     bool isExpr = isVerilogExpression(&op);
-    if (isExpr) {
+    bool isModport = isa<sv::GetModportOp>(op);
+    // If this is an expression emitted inline or unused, it doesn't need a
+    // name. Modport expressions do need a name, but don't need to be emitted.
+    if (isExpr && !isModport) {
```

Comment:
Curious what others think about the updates in `collectNamesEmitDecls` here. To me, treating `sv.interface.instance` and `sv.modport.get` as declarations like this make sense.

By adding this special casing here, it was straightforward to add support for `sv.modport.get`, which is a little like an expression that can't be emitted inline, doesn't need to be emitted out of line, and has a name. I hope it is the right place to put this logic and isn't too much of a kludge.

Reply:
I'm not familiar enough with the expression output so I'd want @lattner to weigh in here.

Reply:
I think the right thing is to treat the instance as a definition, but `sv.modport.get` should be an inline expression.